### ISSUE TITLE
[BUG] Plugin not properly compatible with desugar version 1.2.* (Android 33)

### ISSUE THREAD

**Issue by kerk12** [JulianAssmann/flutter_background#66]
**Describe the bug**
I've recently tried to develop an app that utilizes your plugin. The plugin works great on an emulator w. Android 31, no compatibility issues there. The issue arises with Android 33 based emulators. I'm using the `flutter_local_notification` plugin which utilizes `desugar`. The latest `desugar` version (1.2.2), which is fully compatible w. Android 33 is [incompatible](https://github.com/google/desugar_jdk_libs/blob/master/CHANGELOG.md) with this plugin, as this plugin uses an older Gradle version.

Log: 

```
The Android Gradle plugin supports only Kotlin Gradle plugin version 1.5.20 and higher.
The following dependencies do not satisfy the required version:
project ':flutter_background' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.50
```

**To Reproduce**
Create a new project that utilizes both this plugin and `flutter_local_notifications`. Try to run it in an Android 33-based device. It should crash when the app starts. Also, when upgrading Gradle, it should not be able to be compiled.

**Expected behavior**
The plugin should be able to be compiled on the latest Gradle version.

**Comment by kerk12** [comment]
This is directly related to #65.

**Comment by kingstertime** [comment]
fix please

**Comment by JulianAssmann** [comment]
@kingstertime @kerk12 #65 should be fixed in the latest release (1.2.0). Can you confirm this issue is resolved as well, so I can close it?
